### PR TITLE
Refactor some of the controller & search building

### PR DESCRIPTION
**Pull request by tijmenb**
This PR refactors some code to reduce indirection and improve readability. 

Best reviewed per-commit.

### CODE REVIEW

**Comment by mobaig** [comment]
__Move API clients into Services module__ commit - needs rewording to `instead of doing dependency injection`

File: app/lib/finder_api.rb [hunk 1]
```diff
@@ -11,16 +18,18 @@ def fetch(base_path, params)
 
 private
 
-  def fetch_content_item(base_path)
+  attr_reader :base_path, :filter_params
```
Comment (mobaig) [inline app/lib/finder_api.rb]: Intentional for these to be private?

File: app/lib/search_query_builder.rb [hunk 2]
```diff
@@ -102,10 +100,10 @@ def reject_query
   end
 
   def filter_params
-    @filter_params ||= filter_query_builder.call(
+    @filter_params ||= FilterQueryBuilder.new(
```
Comment (mobaig) [inline app/lib/search_query_builder.rb]: 👏

**Comment by mobaig** [comment]
Looks good 👍 . One of the commits reads a little funny, happy to merge once amended.

**Comment by tijmenb** [comment]
@mobaig updated!

**Pull request merged**